extension/src/experiments/columns/collect/util.ts

```diff
@@ -10,22 +11,35 @@ export const limitAncestorDepth = (
   limit = 5
 ) => {
   const [path, ...rest] = ancestors
-  /*
-    The depth is only limited for the middle of the path array.
-    The first and final layer are excluded, and the
-    concatenated layer itself counts as one; because of this, we must subtract 3
-    from what we want the final layer count to be.
-  */
   const collectedLimit = Number(
     getConfigValue(ConfigKey.EXP_TABLE_HEAD_MAX_LAYERS, limit)
   )
-  const convertedLimit = (collectedLimit >= 3 ? collectedLimit : 3) - 3
-  if (rest.length <= convertedLimit) {
-    return ancestors
+
+  switch (collectedLimit) {
+    case 1:
+      return []
+    case 2:
+      return [
+        buildMetricOrParamPath(ColumnType.PARAMS, ...ancestors).slice(
```

Comment:
[Q] Should `limitAncestorDepth` function handle all `ColumnTypes`? If I pass a metric in here with the limit set to 2 I'll get a Param back out?

Reply:
Created a function for creating the path that takes in separator arguments instead of using `buildMetricOrParamPath`.